client/main.go

```diff
@@ -65,15 +44,17 @@ func NewFullClient(conn grpc.ClientConnInterface) *FullClient {
 type ConnOptions struct {
 	CertURL       string
 	GRPCAuthority string
+	Insecure      bool
 }
 
 func (o *ConnOptions) SetFlags(flagSet *pflag.FlagSet) {
 	flagSet.StringVar(&o.CertURL, "tinkerbell-cert-url", "http://127.0.0.1:42114/cert", "The URL where the certificate is located")
 	flagSet.StringVar(&o.GRPCAuthority, "tinkerbell-grpc-authority", "127.0.0.1:42113", "Link to tink-server grcp api")
+	flagSet.BoolVar(&o.Insecure, "insecure", false, "Run in insecure mode (no TLS)")
 }
 
-func NewClientConn(opt *ConnOptions) (*grpc.ClientConn, error) {
-	resp, err := http.Get(opt.CertURL)
+func fetchCert(url string) (credentials.TransportCredentials, error) {
```

Comment:
I think the function name could be a bit more descriptive. Maybe `grpcCredentialFromCertEndpoint`, or `trustedCertsFromCertEndpoint`.

Either that, or a comment would be helpful.

Reply:
agreed

Reply:
whynotboth.gif

Reply:
food for thought on naming: https://talks.golang.org/2014/names.slide#3